### PR TITLE
Update in Training function and added validate function - need to add more compatible metrics

### PR DESCRIPTION
**Pull request by priyatharsan**
I didnt do the previous changes yet. Thought I would do them together for both PRs altogether

### CODE REVIEW

File: examples/test.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+from vulcanai2 import models, datasets
+
+import torch
+import torch.nn as nn
+import torchvision.transforms as transforms
+from torch.utils.data import DataLoader
+
+
+sys.path.append('../')
+normalize = transforms.Normalize(mean=[x/255.0 for x in [125.3, 123.0, 113.9]],
```
Comment (rfratila) [inline examples/test.py]: why are these 3 values chosen for x?

File: examples/test.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+                                     std=[x/255.0 for x in [63.0, 62.1, 66.7]])
+
+transform = transforms.Compose([transforms.ToTensor(),
+                                transforms.Normalize((0.1307,), (0.3081,))])
```
Comment (rfratila) [inline examples/test.py]: why these normalization values?

**Comment by priyatharsan** [comment]
this branch has been based off in this https://github.com/Aifred-Health/Vulcan2/pull/3
